features/src/main/scala/com/salesforce/op/features/types/Maps.scala

```diff
@@ -291,6 +291,46 @@ object GeolocationMap {
   def empty: GeolocationMap = FeatureTypeDefaults.GeolocationMap
 }
 
+/**
+ * Name representation - map containing information related to a particular name.
+ *
+ * @param value map of keys to values, where keys are one of the following:
+ * - "original"
+ * - "isName"
+ * - "firstName"
+ * - "lastName"
+ * - "gender"
+ */
+class NameStats(value: Map[String, String]) extends TextMap(value) {
+  import NameStats.Keys._
+
+  def isName: Boolean = value.getOrElse(IsNameIndicator, "false") == "true"
+  def isMale: Boolean = value.getOrElse(Gender, "") == "male"
+  def isFemale: Boolean = value.getOrElse(Gender, "") == "female"
+}
+object NameStats {
+  object Keys {
+    val OriginalName = "original"
+    val IsNameIndicator = "isName"
+    val FirstName = "firstName"
+    val LastName = "lastName"
+    val Gender = "gender"
+  }
+  object BooleanStrings {
+    val True = "true"
+    val False = "false"
+  }
+  object GenderStrings {
+    val Male = "male"
+    val Female = "female"
+    val GenderNA = "NA"
+    val GenderNotInferred = "NotInferred"
+  }
+
+  def apply(value: Map[String, String]): NameStats = new NameStats(value)
+  def empty: NameStats = FeatureTypeDefaults.NameStats
+}
+
```

Comment:
please add your feature type to the feature type tests here: https://github.com/salesforce/TransmogrifAI/tree/master/features/src/test/scala/com/salesforce/op/features/types